js/controller.js

```diff
@@ -1214,9 +1217,10 @@ OO.plugin('Html5Skin', function(OO, _, $, W) {
       this.state.seeking = false;
       if (this.state.queuedPlayheadUpdate) {
         OO.log('popping queued update');
-        this.skin.updatePlayhead.apply(this.skin, this.state.queuedPlayheadUpdate);
-        this.state.queuedPlayheadUpdate = null;
-        this.renderSkin();
+        this.skin.updatePlayhead.apply(this.skin, this.state.queuedPlayheadUpdate).then(() => {
+          this.state.queuedPlayheadUpdate = null;
+          this.renderSkin();
```

Comment:
Changing the state of the skin (updatePlayhead doing this) actually should trigger `skin.render` automatically. What would happen if we remove `this.renderSkin` following after `this.skin.updatePlayhead` call? Should continue working like a charm.

Reply:
Yes, it looks like you are right, so I removed all this.renderSkin from 'then' . But I did not tested all cases where we called this.renderSkin .